api/tests/core/external/sendinblue_test.py

```diff
@@ -359,3 +366,90 @@ def test_make_update_request(self):
+        offer1 = offer_factories.OfferFactory(
+            name="Offer 1",
+            description="Offer 1 description",
+            product=product1,
+            url="https://example.com/offer1",
+            subcategoryId="CINE_PLEIN_AIR",
+        )
+        offer2 = offer_factories.OfferFactory(
+            name="Offer 2",
+            subcategoryId="SUPPORT_PHYSIQUE_FILM",
+            description=None,
+        )
+        offers = [offer1.id, offer2.id]
+        with assert_num_queries(1):
+            recommendations = get_offers_recommendations_from_ids(offers)
+        assert recommendations == {
+            offer1.id: Recommendation(
+                url="https://webapp-v2.example.com/offer/" + str(offer1.id),
+                name="Offer 1",
+                description="Offer 1 description",
+                image_url=offer1.image.url,
+                external_url="https://example.com/offer1",
+            ),
+            offer2.id: Recommendation(
+                url="https://webapp-v2.example.com/offer/" + str(offer2.id),
+                name="Offer 2",
+            ),
+        }
+
+    def test_update_contact_recommendations(self):
+        offer1 = offer_factories.OfferFactory(
+            name="Offer 1",
+            description="Offer 1 description",
+            url="https://example.com/offer1",
+            subcategoryId="CINE_PLEIN_AIR",
+        )
+        offer2 = offer_factories.OfferFactory(
+            name="Offer 2",
+            subcategoryId="SUPPORT_PHYSIQUE_FILM",
+        )
+        recommendations = get_offers_recommendations_from_ids([offer1.id, offer2.id])
+        user = UserFactory()
+        update_contact_recommendations(user, recommendations)
+        assert sendinblue_testing.sendinblue_requests[0] == {
+            "email": user.email,
+            "attributes": {"RECOMMENDATIONS": recommendations},
+            "emailBlacklisted": None,
+        }
+
+    @patch("pcapi.core.external.commands.update_sendingblue_recommendations.get_recommendation_offer_ids")
+    def test_update_contacts_recommendations(self, mock_get_recommendation_offer_ids):
+        offer1 = offer_factories.OfferFactory(
+            name="Offer 1",
+            description="Offer 1 description",
+            url="https://example.com/offer1",
+            subcategoryId="CINE_PLEIN_AIR",
+        )
+        offer2 = offer_factories.OfferFactory(
+            name="Offer 2",
+            subcategoryId="SUPPORT_PHYSIQUE_FILM",
+        )
+        recommendations = get_offers_recommendations_from_ids([offer1.id, offer2.id])
+        users = [UserFactory(), UserFactory()]
+        mock_get_recommendation_offer_ids.return_value = [
+            (users[0], [offer1.id, offer2.id]),
+            (users[1], [offer1.id, offer2.id]),
+        ]
+        with assert_num_queries(3):
+            # 1 call when users[0].id is callled since users[0] is a UserFactory object  the same for users[1] and a last call for the actual request
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏Tu peux écrire sous forme de liste en bullet points, on comprend tout de suite que 1 point = 1 query
```suggestion
            # - call to users[0].id
            # - call to users[1].id, because they are UserFactory objects
            # - the actual request